diff -rup ibus-sayura-1.2.99.20100209/src/engine.c ibus-sayura-1.2.99.20100209_mod/src/engine.c
--- ibus-sayura-1.2.99.20100209/src/engine.c	2010-07-12 20:47:35.000000000 +0530
+++ ibus-sayura-1.2.99.20100209_mod/src/engine.c	2010-07-12 20:47:16.000000000 +0530
@@ -310,7 +310,7 @@ ibus_sinhala_engine_process_key_event (I
     if (c >= 0) /* a consonent is pressed. */
         return ibus_sinhala_handle_vowel_pressed (sinhala, keyval, c);
 
-    if (keyval == IBUS_Shift_L || keyval == IBUS_Shift_L ) {
+    if (keyval == IBUS_Shift_L || keyval == IBUS_Shift_R ) {
         return FALSE;
     }
 
