diff -rup ibus-sayura-1.2.99.20100209/src/engine.c ibus-sayura-1.2.99.20100209_mod/src/engine.c
--- ibus-sayura-1.2.99.20100209/src/engine.c	2010-02-09 10:09:08.000000000 +0530
+++ ibus-sayura-1.2.99.20100209_mod/src/engine.c	2010-06-15 16:25:56.478330264 +0530
@@ -23,9 +23,9 @@ struct _IBusSinhalaEngineClass {
 };
 
 struct {
-	gunichar character;
-	gunichar mahaprana;
-	gunichar sagngnaka;
+	gint character;
+	gint mahaprana;
+	gint sagngnaka;
 	int key;
 } consonents[] = {
 	{0xda4, 0x00, 0x00, IBUS_z},
@@ -72,10 +72,10 @@ struct {
 };
 
 struct {
-	gunichar single0;
-	gunichar double0;
-	gunichar single1;
-	gunichar double1;
+	gint single0;
+	gint double0;
+	gint single1;
+	gint double1;
 	int key;
 } vowels[] = {
 	{0xd85, 0xd86, 0xdcf, 0xdcf, IBUS_a},
@@ -207,7 +207,7 @@ ibus_sinhala_engine_init (IBusSinhalaEng
     sinhala->prop_list = ibus_prop_list_new ();
     g_object_ref_sink (sinhala->prop_list);
     ibus_prop_list_append (sinhala->prop_list,  sinhala->sinhala_mode_prop);
-    sinhala->buffer = g_array_new (TRUE, TRUE, sizeof(gunichar *));
+    sinhala->buffer = g_array_new (TRUE, TRUE, sizeof(gint));
 }
 
 static GObject*
